sources/bing_webmaster/__init__.py

```diff
@@ -0,0 +1,92 @@
+"""
+A source loading history of organic search traffic from Bing Webmaster API
+See documentation: https://learn.microsoft.com/en-us/dotnet/api/microsoft.bing.webmaster.api.interfaces.iwebmasterapi?view=bing-webmaster-dotnet
+"""
+
+import time
+from typing import Iterable, Iterator, List, Sequence
+
+import dlt
+from dlt.common.typing import DictStrAny, DictStrStr
+from dlt.sources import DltResource
+
+from .helpers import get_stats_with_retry, parse_response
+
+
+@dlt.source(name="bing_webmaster")
+def source(
+    site_urls: List[str] = None, site_url_pages: Iterable[DictStrStr] = None
+) -> Sequence[DltResource]:
+    """
+    A dlt source for the Bing Webmaster api.
+    It groups resources for the APIs which return organic search traffic statistics
+    Args:
+        site_urls: List[str]: A list of site_urls, e.g, ["dlthub.com", "dlthub.de"]. Use this if you need the weekly traffic per site_url and page
+        site_url_pages: Iterable[DictStrStr]: A list of pairs of site_url and page. Use this if you need the weekly traffic per site_url, page, and query
+    Returns:
+        Sequence[DltResource]: A sequence of resources that can be selected from including page_stats and page_query_stats.
+    """
+    return (
+        page_stats(site_urls),
+        page_query_stats(site_url_pages),
+    )
+
+
+@dlt.resource(
+    write_disposition="merge",
```

Comment:
is it possible to make it incremental (does the response have a "created_at" or similar key?)? you use merge mode and without incremental loading it will work very inefficient

Reply:
> "merge" is a [type of incremental loading](https://dlthub.com/docs/general-usage/incremental-loading).

It is, I talked only about efficiency =))
Such a weird API, I think you can keep it like this, looks like there is no another solution. When you request data there is no params like "since" or "end_date" or similar?

Reply:
Good, thank you for the confirmation!
No, there is neither any filter parameter nor any pagination. They give all the data for 26 weeks for all your thousands of pages in a single response.